packages/devtools_app/test/test_infra/fixtures/networking_app/bin/main.dart

```diff
@@ -0,0 +1,170 @@
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+// ignore_for_file: avoid_print
+
+import 'dart:async';
+import 'dart:convert' show json;
+import 'dart:io' as io;
+
+import 'package:dio/dio.dart';
+import 'package:http/http.dart' as http;
+
+void main() async {
+  final testServer = await _bindTestServer();
+  await _bindControlServer(testServer);
+}
+
+/// Binds a "test" HTTP server to an available port.
+///
+/// This server can receive requests, and responds to them.
+Future<io.HttpServer> _bindTestServer() async {
+  final server = await io.HttpServer.bind(io.InternetAddress.loopbackIPv4, 0);
+  server.listen((request) {
+    request.response.write('fallthrough');
+    if (request.uri.path.contains('complete/')) {
+      request.response.close();
+    }
+  });
+  return server;
+}
+
+/// Binds a "control" HTTP server to an available port.
+///
+/// This server has an HTTP client, and can receive commands for that client to
+/// send requests to the "test" HTTP server.
+Future<io.HttpServer> _bindControlServer(io.HttpServer testServer) async {
+  final client = _HttpClient(testServer.port);
+
+  final server = await io.HttpServer.bind(io.InternetAddress.loopbackIPv4, 0);
+  print(json.encode({'controlPort': server.port}));
+  server.listen((request) {
+    request.response.headers.add('Access-Control-Allow-Origin', '*');
+    request.response.headers.add(
+      'Access-Control-Allow-Methods',
+      'POST,GET,DELETE,PUT,OPTIONS',
+    );
+    final path = request.uri.path;
+    final hasBody = path.contains('/body/');
+    request.response.statusCode = 200;
+    request.response.write('>$path<');
+
+    if (path.startsWith('/get/')) {
+      client.get();
+    } else if (path.startsWith('/post/')) {
+      client.post(hasBody: hasBody);
+    } else if (path.startsWith('/put/')) {
+      client.put(hasBody: hasBody);
+    } else if (path.startsWith('/delete/')) {
+      client.delete(hasBody: hasBody);
+    } else if (path.startsWith('/dio/get/')) {
+      client.dioGet();
+    } else if (path.startsWith('/dio/post/')) {
+      client.dioPost(hasBody: hasBody);
+    } else if (path.startsWith('/packageHttp/post/')) {
+      client.packageHttpPost(hasBody: hasBody);
+    } else if (path.startsWith('/packageHttp/postStreamed/')) {
+      client.packageHttpPostStreamed();
+    }
+    request.response.close();
+  });
+  return server;
+}
+
+// TODO WebSocket
+// TODO package:http - BrowserClient - https://pub.dev/documentation/http/latest/browser_client/BrowserClient-class.html
+// TODO cupertino_http - https://pub.dev/packages/cupertino_http
+// TODO cronet_http - https://pub.dev/packages/cronet_http
+// TDOO fetch_client? https://pub.dev/packages/fetch_client
+
+class _HttpClient {
+  _HttpClient(int testServerPort)
+    : _uri = Uri.http('127.0.0.1:$testServerPort', '/');
+
+  final Uri _uri;
+
+  final client = io.HttpClient();
+
+  final _dio = Dio();
```

Comment:
do any of these http clients need to be closed at some point?

Reply:
Hmm, I haven't seen a need to close them. Is there a point in the test lifecycle that I can? Like, in tearDown I could send a request to exit the process.